cmd/http/listener.go

```diff
@@ -120,11 +120,15 @@ func getResourceHost(bufConn *BufConn, maxHeaderBytes int) (resource string, met
 
 		if method == "" && resource == "" {
 			if i := strings.IndexByte(tokens[0], ' '); i == -1 {
-				return "", "", "", fmt.Errorf("malformed HTTP request %s, from %s", tokens[0], bufConn.LocalAddr())
+				return "", "", "", fmt.Errorf("malformed HTTP request, from '%s'", bufConn.LocalAddr())
 			}
 			httpTokens := strings.SplitN(tokens[0], " ", 3)
 			if len(httpTokens) < 3 {
-				return "", "", "", fmt.Errorf("malformed HTTP request %s, from %s", tokens[0], bufConn.LocalAddr())
+				return "", "", "", fmt.Errorf("malformed HTTP request, from '%s'", bufConn.LocalAddr())
+			}
+			if !isHTTPMethod(httpTokens[0]) {
```

Comment:
This function is called after fetching HTTP method.  This is not required

Reply:
This was simply the old code, are you saying we should let Go HTTP handle this?

Reply:
The reason to validate the method is one can generate something like this

```
curl -X XXX http://localhost:9000
``` 

this can clobber up our stats @balamurugana - i just moved the code from outside to inside.